extensions/common/vault/vault-aws/src/main/java/org/eclipse/edc/vault/aws/AwsSecretsManagerVaultDefaultSanitationStrategy.java

```diff
@@ -25,34 +26,41 @@ public AwsSecretsManagerVaultDefaultSanitationStrategy(Monitor monitor) {
 
     /**
      * Many-to-one mapping from all strings into set of strings that only contains valid AWS Secrets Manager key names.
-     * The implementation replaces all illegal characters with '_' and attaches the hash code of the original string to
-     * minimize the likelihood of key collisions.
+     * The implementation replaces all illegal characters with '_' and attaches the hash code of the original string,
+     * when illegal characters are replaced, to minimize the likelihood of key collisions.
      *
      * @param originalKey any key
      * @return Valid AWS Secrets Manager key
      */
     @Override
     public String sanitizeKey(String originalKey) {
         var key = originalKey;
-        if (originalKey.length() > 500) {
+        boolean originalKeyReplaced = false;
+
+        if (originalKey.length() > 512) {
+            //Substring to 500, to add the 12 char hashcode
             key = originalKey.substring(0, 500);
```

Comment:
I didn't get this 500/512 thing, looking at the tests seems like it's a sort of size limit for the key size, eventually it could be extracted as a constant called `KEY_SIZE_LIMIT` or something like that. 
The comment is outdated (please adapt the code to being able to talk for itself)